lib/sidekiq/job_metadata_middleware.rb

```diff
@@ -0,0 +1,12 @@
+# frozen_string_literal: true
+
+module Sidekiq
+  class JobMetadataMiddleware
+    def call(job_instance, job_payload, _queue)
+      return unless job_instance.is_a?(Sidekiq::JobMetadata)
```

Comment:
Followed the pattern in the `RetryMonitoring` class to use include `JobMetadata` in a job to expose the job data.